eclair-core/src/main/scala/fr/acinq/eclair/Setup.scala

```diff
@@ -220,11 +220,11 @@ class Setup(datadir: File,
       routerTimeout = after(FiniteDuration(config.getDuration("router.init-timeout").getSeconds, TimeUnit.SECONDS), using = system.scheduler)(Future.failed(new RuntimeException("Router initialization timed out")))
       _ <- Future.firstCompletedOf(routerInitialized.future :: routerTimeout :: Nil)
 
+      chaindir = new File(datadir, chain)
```

Comment:
There must be a better way to do this than to redefine the directory here. There was even a `TODO` comment that you missed when you moved it.